compiler-rt/lib/sanitizer_common/sanitizer_common.h

```diff
@@ -290,6 +290,17 @@ bool SetEnv(const char *name, const char *value);
 
 u32 GetUid();
 void ReExec();
+
+#if defined(__ELF__) && !SANITIZER_FUCHSIA
```

Comment:
Why does Fuchsia not want this checking?

Reply:
`compiler-rt/lib/asan/asan_fuchsia.cpp` has `void *AsanDoesNotSupportStaticLinkage() { return nullptr; }`, which opts out the check. I am not familiar with Fuchsia, but it doesn't seem to use interceptors, so technically static linking is supported, unlike other ELF OSes. 

@petrhosek

Reply:
The reference is intentionally in an inline function in the `RTSanitizerCommon` library, so that if the function is not called, there is no such check. If we move it into `.cpp`, programs linking in ubsan/xray will fail for `-static`.

I think we can make ubsan standalone runtime work for `-static`: https://github.com/llvm/llvm-project/pull/80943
At the very least, we should not let `-fsanitize-minimal-runtime` fail for `-static`.

Reply:
Moved to `compiler-rt/lib/interception/interception.h` to make it clear `-static` unsupportness is an interception-related requirement.